src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.ReadHelper.cs

```diff
@@ -78,65 +78,46 @@ public partial class JsonTypeInfo<T>
             }
         }
 
-        private JsonTypeInfo<Queue<T>>? _asuncEnumerableQueueTypeInfo;
-        internal IAsyncEnumerable<T> DeserializeAsyncEnumerable(Stream utf8Json, CancellationToken cancellationToken)
-        {
-            Debug.Assert(IsConfigured);
-
-            JsonTypeInfo<Queue<T>>? queueTypeInfo = _asuncEnumerableQueueTypeInfo;
-            if (queueTypeInfo is null)
-            {
-                var queueConverter = new QueueOfTConverter<Queue<T>, T>();
-                queueTypeInfo = new JsonTypeInfo<Queue<T>>(queueConverter, Options)
-                {
-                    CreateObject = static () => new Queue<T>(),
-                    ElementTypeInfo = this,
-                    NumberHandling = Options.NumberHandling,
-                };
+        // Creating a queue JsonTypeInfo from within the DeserializeAsyncEnumerable method
+        // triggers polymorphic recursion warnings from the AOT compiler so we instead
+        // have the callers do it for us externally (cf. https://github.com/dotnet/runtime/issues/84922)
+        internal JsonTypeInfo<Queue<T>>? AsyncEnumerableQueueTypeInfo;
```

Comment:
```suggestion
        internal JsonTypeInfo<Queue<T>>? _asyncEnumerableQueueTypeInfo;
```
Nit: Naming convention for fields.